unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -231,5 +287,54 @@ export function QuerySuggestPreviewTest() {
         }, test.cmp.options.executeQueryDelay);
       });
     });
+    describe('Analytics', () => {
+      function getAResult(done) {
+        const previewContainer = $$(suggestionContainer.el).find('.coveo-preview-results > .CoveoResult');
+        if (!previewContainer) {
+          done.fail('No result to click. Impossible validate the analytics');
+        }
+        return previewContainer;
+      }
+
+      it('it log an analytics with the appropriate event', done => {
+        const suggestion = 'test';
+        triggerQuerySuggestHover(suggestion);
+        setTimeout(() => {
+          const previewContainer = getAResult(done);
+          previewContainer.click();
+          expect(test.cmp.usageAnalytics.logCustomEvent).toHaveBeenCalledWith(
+            analyticsActionCauseList.clickQuerySuggestPreview,
+            jasmine.objectContaining({
+              suggestion,
+              displayedRank: 0
+            }),
+            previewContainer
+          );
+          done();
+        }, test.cmp.options.executeQueryDelay);
+      });
+
+      it(`it log an analytics with the appropriate event,
```

Comment:
This test simulates when we hover `200ms` (the minimum time) on a suggestion to lunch a query. 
After, we hover `100ms` on a suggestion (which is not enough time to lunch a query).
After we click on a result. 
It should log analytics with the first suggestion hovered.